pkg/server/endpoints/registration/handler.go

```diff
@@ -525,6 +588,18 @@ func (h *Handler) prepareRegistrationEntry(entry *common.RegistrationEntry, forU
 	return entry, nil
 }
 
+func (h *Handler) startCall(ctx context.Context, key string, keyn ...string) (*telemetry.CallCounter, error) {
```

Comment:
Thank you for all the tests around authorization.

One thing that worries me a little bit here is the responsibility carried during code review to make sure that any new methods added to this handler call `startCall`. It seems like it could be easy to overlook, especially when the call kind of looks like a telemetry call.

I noticed you are using interceptors in the test suite... is authorization a good use case there? i.e. install interceptors that are aware of the available services+methods and their associated authz logic... that way, any new endpoints that don't have a policy installed will be denied, and we won't fail open... I think at the very least, we should change the name of this method or make two explicit calls, one of which is `authorizeCaller`... thoughts?